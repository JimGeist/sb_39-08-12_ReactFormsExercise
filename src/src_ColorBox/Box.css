.Box {
    margin: 5px 0 5px 5px;
    border-radius: 5px;
    border-style: solid;
    border-color: black;
    border-width: 1px;
    float:left;
}
.Box-BtnDelete {
    margin: 5px 5px 0 5px;
}