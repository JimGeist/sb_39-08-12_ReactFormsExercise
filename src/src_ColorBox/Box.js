import React from "react";

import './Box.css';

function Box({ id, width, height, bgColor, deleteFx }) {

    const deleteBox = () => {
        deleteFx(id);
    }

    return (
        <div className="Box">
            <div className="Box"
                style={{ width: +width, height: +height, backgroundColor: bgColor }}
                data-testid="^Color Box^" >
            </div>
            <button className="Box-BtnDelete" onClick={deleteBox}>X</button>
        </div>
    );
}

export default Box;
