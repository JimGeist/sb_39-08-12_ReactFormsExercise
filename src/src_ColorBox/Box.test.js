import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

// Smoke Test -- does Box render?
it("renders the Box component without crashing", () => {
    render(<Box />);
});

// Box snapshot tests
it("Box component without props matches snapshot", () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});

it("Box component with width, height, and color background matches snapshot", () => {
    const { asFragment } = render(<Box width={100} height={100} bgColor="black" />);
    expect(asFragment()).toMatchSnapshot();
});
