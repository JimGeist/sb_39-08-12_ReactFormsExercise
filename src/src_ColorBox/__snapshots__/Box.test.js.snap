// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component with width, height, and color background matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box"
      data-testid="^Color Box^"
      style="width: 100px; height: 100px; background-color: black;"
    />
    <button
      class="Box-BtnDelete"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`Box component without props matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box"
      data-testid="^Color Box^"
    />
    <button
      class="Box-BtnDelete"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
